echopop/survey.py

```diff
@@ -213,20 +216,299 @@ def stratified_analysis(
+        """
+
+        # Validate "variable" input
+        if variable not in ["biomass", "abundance"]:
+            raise ValueError(
+                f"The user input for `variable` ({variable}) is invalid. Only `variable='biomass'` "
+                f"and `variable='abundance'` are valid inputs for the `fit_variogram()` method."
+            )
+
+        # Initialize Survey-class object
+        self.analysis.update({"variogram": {}})
+
+        # Parameterize analysis settings that will be applied to the variogram fitting and analysis
+        self.analysis["settings"].update(
+            {
+                "variogram": {
+                    "azimuth_range": azimuth_range,
+                    "fit_parameters": fit_parameters,
+                    "model": model,
+                    "standardize_coordinates": standardize_coordinates,
+                    "stratum_name": self.analysis["settings"]["transect"]["stratum_name"],
+                    "variable": variable,
+                    "verbose": verbose,
+                }
+            }
+        )
+
+        # Append `kriging_parameters` to the settings dictionary
+        if standardize_coordinates:
+            self.analysis["settings"]["variogram"].update(
+                {"kriging_parameters": self.input["statistics"]["kriging"]["model_config"]}
+            )
+
+        # Create a copy of the existing variogram settings
+        variogram_parameters = self.input["statistics"]["variogram"]["model_config"].copy()
+
+        # Update the `variogram_parameters` dictionary based on user inputs
+        # ---- Model name
+        variogram_parameters["model"] = model
+        # ---- Nugget
+        if nugget is not None:
+            variogram_parameters["nugget"] = nugget
+        # ---- Sill
+        if sill is not None:
+            variogram_parameters["sill"] = sill
+        # ---- Correlation range
+        if correlation_range is not None:
+            variogram_parameters["correlation_range"] = correlation_range
+        # ---- Hole effect range
+        if hole_effect_range is not None:
+            variogram_parameters["hole_effect_range"] = hole_effect_range
+        # ---- Decay power
+        if decay_power is not None:
+            variogram_parameters["decay_power"] = decay_power
+        # ---- Semivariance enhancement
+        if enhance_semivariance is not None:
+            variogram_parameters["enhance_semivariance"] = enhance_semivariance
+        # ---- Lag resolution
+        if lag_resolution is not None:
+            variogram_parameters["lag_resolution"] = lag_resolution
+        # ---- Max range
+        if max_range is not None:
+            variogram_parameters["range"] = max_range
+        # ---- Number of lags
+        variogram_parameters["n_lags"] = n_lags
+        # ---- Azimuth range
+        variogram_parameters["azimuth_range"] = azimuth_range
+        # ---- Force lag-0
+        variogram_parameters["force_lag_zero"] = force_lag_zero
```

Comment:
I think we do something similar to [what i commented on on the optimization params](https://github.com/OSOceanAcoustics/echopop/pull/251/files#r1720595160), to require users to set the parameters in a dict using key-value pairs, have the default values in a dict to initialize `variogram_parameters` first, and then update `variogram_parameters` with user inputs via the `.update` function. Like:
```python
DEFAULT_VPARAM_DICT = {
    "param1": value1, ...
}

... # code blocks in the middle

# assumes variogram_params below is a user input argument
v_params_out = DEFAULT_VPARAM_DICT
v_params_out["model"] = model
if not isinstance(variogram_params, dict):  # make sure input is a dict
    raise ValueError("this should be a dict")
if variogram_params:  # if not None
    # update using all input params at once
    v_params_out.update(variogram_params)
```